hazelcast/src/main/java/com/hazelcast/sql/impl/client/SqlExecuteMessageTask.java

```diff
@@ -92,7 +93,9 @@ protected ClientMessage encodeResponse(Object response) {
     }
 
     protected ClientMessage encodeException(Throwable throwable) {
-        nodeEngine.getSqlService().getInternalService().getClientStateRegistry().closeOnError(parameters.queryId);
+        // exception can be thrown before parameters are decoded
+        QueryId queryId = parameters != null ? parameters.queryId : decodeClientMessage(clientMessage).queryId;
+        nodeEngine.getSqlService().getInternalService().getClientStateRegistry().closeOnError(queryId);
```

Comment:
I wouldn't decode the client message here - it is decoded somewhere else, and if it wasn't decoded, we certainly don't have the cursor in the registry yet. I think we should ignore the `closeOnError` call in case `parameters == null`.